### ISSUE TITLE
There is no document formatter for 'html'-files installed

### ISSUE THREAD

**Issue by DarkLite1** [Microsoft/vscode#57576]
Version: 1.27.0-insider (user setup)
Commit: 98c83caea995f152003c264658c33186d164dfa8
Date: 2018-08-30T05:19:32.583Z
Electron: 2.0.7
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

When trying to format a HTML document. it get the following message:

> There is no document formatter for 'html'-files installed

UserSettings.json
```
{
    "editor.formatOnType": true,
    "editor.formatOnPaste": true,
    "editor.wordWrap": "on",
    "terminal.integrated.shell.windows": "C:\\Windows\\sysnative\\WindowsPowerShell\\v1.0\\powershell.exe",
    "window.zoomLevel": 0,
    "git.ignoreMissingGitWarning": true,
    "editor.fontSize": 12
}
```

WorkspaceSettings.json
```
{
	"folders": [
		{
			"path": "s:\\Prod"
		},
		{
			"path": "s:\\Test\\Brecht"
		},
		{
			"path": "c:\\Program Files\\WindowsPowerShell\\Modules"
		}
	],
	"settings": {
		"liveServer.settings.multiRootWorkspaceName": "Brecht"
	}
}
```

Default user settings says:
```
// Enable/disable default HTML formatter.
  "html.format.enable": true,
```

**Comment by aeschli** [comment]
Can you set `"html.trace.server": "verbose"` and open the 'HTML Language Server' output tab and look our for the following info:
```
[Trace - 2:43:57 PM] Received request 'client/registerCapability - (1)'.
Params: {
    "registrations": [
        {
            "id": "f99847f6-39fe-4660-adfd-bf65f86290ea",
            "method": "textDocument/rangeFormatting",
            "registerOptions": {
                "documentSelector": [
                    {
                        "language": "html"
                    },
                    {
                        "language": "handlebars"
                    }
                ]
            }
        }
    ]
}
```

**Comment by aeschli** [comment]
Also try to run without extensions.

**Comment by DarkLite1** [comment]
Thank you for the help. A bit stuck here, can you tell me how to open the 'HTML Language Server' output tab?

**Comment by DarkLite1** [comment]
When I go to the HTML document. go to the bottom and select the tab `Output` with `HTML Language Server`, and then ask in the palette to `Format document` nothing is displayed in the `Output` tab and the pop `There is no document formatter for 'html'-files installed` appears..

**Comment by DarkLite1** [comment]
Had to restart, here is what I got:

```
[Trace - 2:55:10 PM] Received request 'client/registerCapability - (0)'.
Params: {
    "registrations": [
        {
            "id": "2e0da313-e708-4d12-9682-f6aac75feed3",
            "method": "workspace/didChangeWorkspaceFolders",
            "registerOptions": {}
        }
    ]
}


[Trace - 2:55:10 PM] Sending response 'client/registerCapability - (0)'. Processing request took 0ms
No result returned.


[Trace - 2:55:10 PM] Received request 'client/registerCapability - (1)'.
Params: {
    "registrations": [
        {
            "id": "423625c6-a993-45a5-92d9-a0fe28c2dab6",
            "method": "textDocument/rangeFormatting",
            "registerOptions": {
                "documentSelector": [
                    {
                        "language": "html"
                    },
                    {
                        "language": "handlebars"
                    }
                ]
            }
        }
    ]
}


[Trace - 2:55:10 PM] Sending response 'client/registerCapability - (1)'. Processing request took 0ms
No result returned.
```

**Comment by aeschli** [comment]
@jrieken This is a mystery bug that pops up every month or two. The formatter is is very likely correctly registered, but can not be found by the action

**Comment by aeschli** [comment]
@DarkLite1 If you have time and are familiar with the dev tools: This would help getting to the core of it:
- open the dev tools (`Developer: Toggle Developer Tools`)
- in the source tab, press Ctrl + P and select 'format.ts`
![image](https://user-images.githubusercontent.com/6461412/44853324-ee085d00-ac65-11e8-8e2d-cd62d14e2266.png)
- set a breakpoint on line 34
- invoke `Format document`
- breakpoint should be reached
 
![image](https://user-images.githubusercontent.com/6461412/44853385-20b25580-ac66-11e8-8abc-0ebb8c251cc0.png)
- look at the values of `providers` (Probably empty in your case) and DocumentRangeFormattingEditProviderRegistry
- in the hover of DocumentRangeFormattingEditProviderRegistry, move the mouse over `_entries`, open context meny and run 'Store as global variable`
![image](https://user-images.githubusercontent.com/6461412/44853595-aafab980-ac66-11e8-93be-1d698aac560f.png)
- that prints to the console and it would be good to know how many entries there are.
In my case in get something like that:
![image](https://user-images.githubusercontent.com/6461412/44853706-ec8b6480-ac66-11e8-994f-3bbc973654d9.png)

**Comment by jrieken** [comment]
@DarkLite1 What kind of file is that? Can you right click its title, select 'Copy Path', and paste the path here? Is this with VS Live Share?

**Comment by DarkLite1** [comment]
@jrieken  The Path is `S:\Test\Brecht\Set permissions\Settings detail.html`

@aeschli I needed to restart VS Code for a project here, and then I tried again after a few minutes. Result, now it works. You still want me to do the dev tools test thing?

It's really weird, because I haven't changed anything at all.

**Comment by jrieken** [comment]
> @jrieken The Path is S:\Test\Brecht\Set permissions\Settings detail.html. I don't know what VS Live Share is...

Thanks, question answered.

**Comment by aeschli** [comment]
> You still want me to do the dev tools test thing?
only when you see the problem... ;-(

**Comment by DarkLite1** [comment]
Shame, I really wanted to get to the bottom of this. I'll let you know when I have it again and try your steps. Thanks anyhow for the great help, much appreciated.

**Comment by pcostarg** [comment]
I had this issue also.

I believe that it happened after I uninstalled the system wide VS Code, after installing the User version.

I fixed it by installing again the user version.

**Comment by aeschli** [comment]
fixed by #71652
